pkg/apis/k0s.k0sproject.io/v1beta1/network.go

```diff
@@ -47,59 +55,74 @@ type Network struct {
 // DefaultNetwork creates the Network config struct with sane default values
 func DefaultNetwork() *Network {
 	return &Network{
-		PodCIDR:       "10.244.0.0/16",
-		ServiceCIDR:   "10.96.0.0/12",
-		Provider:      "kuberouter",
-		KubeRouter:    DefaultKubeRouter(),
-		DualStack:     DefaultDualStack(),
-		KubeProxy:     DefaultKubeProxy(),
-		ClusterDomain: "cluster.local",
+		PodCIDR:               "10.244.0.0/16",
+		ServiceCIDR:           "10.96.0.0/12",
+		Provider:              "kuberouter",
+		KubeRouter:            DefaultKubeRouter(),
+		DualStack:             DefaultDualStack(),
+		KubeProxy:             DefaultKubeProxy(),
+		NodeLocalLoadBalancer: DefaultNodeLocalLoadBalancer(),
+		ClusterDomain:         "cluster.local",
 	}
 }
 
 // Validate validates all the settings make sense and should work
 func (n *Network) Validate() []error {
+	if n == nil {
+		return nil
+	}
+
 	var errors []error
-	if n.Provider != "calico" && n.Provider != "custom" && n.Provider != "kuberouter" {
-		errors = append(errors, fmt.Errorf("unsupported network provider: %s", n.Provider))
+
```

Comment:
may we could utilize early return here, instead of trying to collect all errors in one run

Reply:
This pattern is the standard in the Kubernetes code base, i.e. all the resource validation methods return error lists. I think this is convenient for users. When you have three errors in your resource, it's easier if all three errors are reported in one go. Otherwise you end up fixing the first, retry, get a new error, fix the second, retry, ...